web/server/web_client_cache.c

```diff
@@ -36,6 +48,12 @@ static void web_client_free(struct web_client *w) {
     buffer_free(w->response.data);
     freez(w->user_agent);
     freez(w);
+#ifdef ENABLE_HTTPS
+    if ( w->ssl.conn){
```

Comment:
Got a SEGFAULT here. I just had netdata use a unix socket and sent a request from nginx with SSL enabled on the server.

Reply:
I tested on Slackware current and Debian 9.9 and I could not see the same problem, so I decided to read the official documentation of the library to understand what was happening. I saw that a possible change that I wrote last night could create this problem and I changed it, please, try again when you have a time. I will also continue testing on my side with different environments.

Reply:
Same thing. 
```
NETDATA_SYSTEM_OS_NAME="Manjaro Linux"
NETDATA_SYSTEM_OS_ID=manjaro
NETDATA_SYSTEM_OS_ID_LIKE=arch
NETDATA_SYSTEM_OS_VERSION=18.0.4
NETDATA_SYSTEM_OS_VERSION_ID=unknown
NETDATA_SYSTEM_OS_DETECTION=Mixed
NETDATA_SYSTEM_KERNEL_NAME=Linux
NETDATA_SYSTEM_KERNEL_VERSION=4.19.32-1-MANJARO
NETDATA_SYSTEM_ARCHITECTURE=x86_64
```

Reply:
It doesn't always happen. Where id did, it looked like poll_close_fd was called twice in the same thread. Will do a few more tests to see if I can find the cause.

Reply:
During my tests yesterday, It never happened, I am doing a review in the parts of the code where I am managing the memory to check this specific problem;

Reply:
I sent you the valgrind output after a segfault

Reply:
This was already fixed and the newest with stream looks like it is solved too!